src/main/java/com/onarandombox/MultiverseCore/utils/result/Attempt.java

```diff
@@ -0,0 +1,174 @@
+package com.onarandombox.MultiverseCore.utils.result;
+
+import com.onarandombox.MultiverseCore.utils.message.Message;
+import com.onarandombox.MultiverseCore.utils.message.MessageReplacement;
+
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+public interface Attempt<T, F extends FailureReason> {
+
+    static <T, F extends FailureReason> Attempt<T, F> success(T value) {
+        return new Success<>(value);
+    }
+
+    static <T, F extends FailureReason> Attempt<T, F> failure(
+            F failureReason, MessageReplacement... messageReplacements) {
+        return new Failure<>(failureReason, Message.of(failureReason, "Failed!", messageReplacements));
+    }
+
+    static <T, F extends FailureReason> Attempt<T, F> failure(F failureReason, Message message) {
+        return new Failure<>(failureReason, message);
+    }
+
+    T get();
+
+    F getFailureReason();
+
+    Message getFailureMessage();
+
+    default boolean isSuccess() {
+        return this instanceof Success;
+    }
+
+    default boolean isFailure() {
+        return this instanceof Failure;
+    }
+
+    default Attempt<T, F> peek(Consumer<T> consumer) {
+        if (this instanceof Success) {
+            consumer.accept(get());
+        }
+        return this;
+    }
+
+    default <N> Attempt<N, F> map(Function<T, Attempt<N, F>> mapper) {
```

Comment:
I think this should be `<U> Attempt<U, F> map(Function<? extends T, ? extends U> mapper)`